### ISSUE TITLE
More general Vowpal Wabbit parser

### ISSUE THREAD

**Issue by romovpa** [bigartm/bigartm#180]
Improvement #161 is great, but..

To support parsing VW dataset out of the box only one thing is missing: VW input are typically for the classification or regression task and has the following format (see [details](https://github.com/JohnLangford/vowpal_wabbit/wiki/Input-format)):
```
[Label] [Importance [Tag]]|Namespace Features ....
```

For example:
```
+1 0.2 'document123 |words sdfsdfsf sdfsdf ...
```

Now we use string before first pipe as an item title. To be more correct, I propose to use Tag field as a title if the input have a prime symbol `'` (thus have the Tag field). And use entire sting before the first pipe as a title otherwise (current behaviour):
```
document123 |words sdfsdf sdfsdf ...
```

The former case is useful when you have VW classification/regression dataset and don't want to transform it any way, but the latter case is more convenient to use with datasets aimed to topic modeling only.

**Comment by sashafrey** [comment]
Thanks for a suggestion! I agree, it sounds reasonable.

Any ideas where to put class label and weight?
- Item weights is something we may introduce in BigARTM via new field ("optional float Item.weight"),  and multiply all n_wd counters for this document by the weight. It defaults to 1.0f, 
- Class label appears to be just another modality. Should I put it into some hard-coded modality? For example into "@label" -- similar to "@default_class" string used for default modality).

**Comment by romovpa** [comment]
As I know, VW-LDA simply ignores label and weight.

Item weights sounds good, but I can't imagine clear use case.

Class `@label` — genious idea! =) But to not confuse users, we need to warn user about this modality because it will affect the entire topic model. Also, can we switch modalities off (ignore them)?

**Comment by sashafrey** [comment]
Yes, one can define which modalities to use in ModelConfig.class_id (repeated field). Empty field imply "use all modalities". This is not exactly "switching off some modalities", but rather you need to know the entire list of modalities to use. But I assume this is not a problem as you need to know the list of modalities anyway (to configure scores, retrieve topic model, etc).

So then we go for ``@label`` modality (plust add a big big "Warning" in the documentation of VW-parser). I'll do the change asap :)

**Comment by sashafrey** [comment]
I looked at the description [here](https://github.com/JohnLangford/vowpal_wabbit/wiki/Input-format), and there are interesting details regarding space before or after pipe |. However, by looking at the original source code [here](https://github.com/JohnLangford/vowpal_wabbit/blob/master/vowpalwabbit/parse_example.cc) is really hard to say what exactly is going on in the parser.
I'll continue looking into this next week.
